### ISSUE TITLE
How to solve UserWarning: categorical_feature in Dataset is overrided.

### ISSUE THREAD

**Issue by elzizyel** [microsoft/LightGBM#2445]
## Environment info

Operating System: macOS

CPU/GPU model: CPU 

C++/Python/R version: Python

LightGBM version or commit hash: 2.1.0

## Training Set Info
<class 'pandas.core.frame.DataFrame'>
Int64Index: 102759 entries, 0 to 102758
Data columns (total 12 columns):
plate_city                       102759 non-null int64
warehouse_province    102759 non-null int64
warehouse_city             102759 non-null int64
warehouse_type            102759 non-null int64
shop_province               102759 non-null int64
shop_city                        102759 non-null int64
if_gps                              102759 non-null int64
if_pay                              102759 non-null int64
match_hour                    102759 non-null int64
distance                         102759 non-null float64
is_same_province          102759 non-null int64
is_same_city                   102759 non-null int64
dtypes: float64(1), int64(11)
memory usage: 10.2 MB

## Training Code
# Baseline

params = {
    'boosting_type':'gbdt',               
    'metric':'rmse',                                   
}

baseline = lgb.LGBMRegressor(
    boosting_type=params['boosting_type'],
    metric=params['metric']
)

baseline.fit(
    X_train, Y_train,
    eval_set=[(X_train, Y_train),(X_valid, Y_valid)], 
    eval_metric='rmse',
    categorical_feature=cat_feats_ind,
    early_stopping_rounds=300)

## Warning Massage
./python3.5/site-packages/lightgbm/basic.py:1038: UserWarning: categorical_feature in Dataset is overrided. New categorical_feature is [0, 1, 2, 3, 4, 5, 6, 7, 8, 10, 11]
  warnings.warn('categorical_feature in Dataset is overrided. New categorical_feature is {}'.format(sorted(list(categorical_feature))))
